src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -322,6 +323,61 @@ export function VislibVisualizationsPieChartProvider(Private) {
       }
     }
 
+    readableNumber(number) {
+      let tier = Math.log10(Math.abs(number)) / 3 | 0;
+      if(tier === 0) return number;
+      if(tier > 4) tier = 4;
+      const postfix = this.ABBREVIATIONS[tier];
+      const scale = Math.pow(10, tier * 3);
+      const scaled = number / scale;
+      let formatted = String(scaled.toFixed(1));
+      if (/\.0$/.test(formatted)) {
+        formatted = formatted.substr(0, formatted.length - 2);
+      }
+      return formatted + postfix;
+    }
+
+    addDonutDescriptor(svg, width, height) {
+      const self = this;
+      const metricLabel = self.events.handler.data.data
+        .raw.columns.find(function (element) {
+          return element.categoryName === 'metric';
+        }).label;
+      const descriptor = svg.append('text')
+        .style('text-anchor', 'middle')
+        .attr('font-size', function (chartData) {
+          const dividend = Math.min(width, height);
+          let length = 0;
```

Comment:
`const length = chartData.raw ? chartData.raw.columns.length : 0;`

And one `let` less .. :-)